### PR TITLE
Add support for burning ground from Essence of Hysteria

### PR DESCRIPTION
**Pull request by Regisle**
edit : this the 2 lines below are no longer relevant

> Built ontop of #4820
> So most changes are from there, this needs a bit more testing before it can be added, and I will fix commits once the other PR is merged

Burning ground added does not use skill cfg, so modifiers to the skill do not apply but global mods do apply, will correctly override any other burning ground the skill has if its larger, and correctly works with full dps as well

need to test if its correctly isolated from the skill

### CODE REVIEW

**Comment by Nostrademous** [comment]
@Regisle you shouldn't check in code with `ConPrintf()` statements in the code. Will clutter up our console.

**Comment by Regisle** [comment]
> @Regisle you shouldn't check in code with `ConPrintf()` statements in the code. Will clutter up our console.

my bad musta been half asleep, thought I removed those thanks

File: src/Modules/CalcOffence.lua [hunk 1]
```diff
@@ -3990,6 +3990,46 @@ function calcs.offence(env, actor, activeSkill)
...
+			local inc = modDB:Sum("INC", dotTypeCfg, "Damage", damageType.."Damage", isElemental[damageType] and "ElementalDamage" or nil)
+			local more = round(modDB:More(dotTypeCfg, "Damage", damageType.."Damage", isElemental[damageType] and "ElementalDamage" or nil), 2)
+			local mult = modDB:Sum("BASE", dotTypeCfg, "DotMultiplier", damageType.."DotMultiplier")
+			local total = baseDropsBurningGround * (1 + inc/100) * more * (1 + mult/100) * effMult
+			if not output.BurningGroundDPS or output.BurningGroundDPS < total then
+				output.BurningGroundDPS = total
+				output.BurningGroundFromIgnite = false
+			end
+		end
+	end
```
Comment (QuickStick123) [inline src/Modules/CalcOffence.lua:4032]:
```suggestion
	
	local baseDropsBurningGround = modDB:Sum("BASE", nil, "DropsBurningGround")
	if baseDropsBurningGround > 0 then
		if canDeal.Fire then
			local dotCfg = {
				flags = bor(ModFlag.Dot),
				keywordFlags = 0
			}
			local dotTakenCfg = copyTable(dotCfg, true)
			local dotTypeCfg = copyTable(dotCfg, true)
			dotTypeCfg.keywordFlags = bor(dotTypeCfg.keywordFlags, KeywordFlag.FireDot)
			local effMult = 1
			if env.mode_effective then
				local resist = 0
				local takenInc = enemyDB:Sum("INC", dotTakenCfg, "DamageTaken", "DamageTakenOverTime", "FireDamageTaken", "FireDamageTakenOverTime")
				local takenMore = enemyDB:More(dotTakenCfg, "DamageTaken", "DamageTakenOverTime", "FireDamageTaken", "FireDamageTakenOverTime")
				if env.modDB:Flag(nil, "EnemyFireResistEqualToYours") then
					resist = env.player.output.FireResist
				else
					resist = enemyDB:Sum("BASE", nil, "FireResist")
				        local base = resist + enemyDB:Sum("BASE", dotTypeCfg, "ElementalResist")
				        resist = base * calcLib.mod(enemyDB, nil, "FireResist")
				        takenInc = takenInc + enemyDB:Sum("INC", dotTypeCfg, "ElementalDamageTaken")
				end
				resist = m_min(resist, data.misc.EnemyMaxResist)
				effMult = (1 - resist / 100) * (1 + takenInc / 100) * takenMore
			end
			local inc = modDB:Sum("INC", dotTypeCfg, "Damage", "FireDamage", "ElementalDamage")
			local more = round(modDB:More(dotTypeCfg, "Damage", "FireDamage", "ElementalDamage"), 2)
			local mult = modDB:Sum("BASE", dotTypeCfg, "DotMultiplier", "FireDotMultiplier")
			local total = baseDropsBurningGround * (1 + inc/100) * more * (1 + mult/100) * effMult
			if not output.BurningGroundDPS or output.BurningGroundDPS < total then
				output.BurningGroundDPS = total
				output.BurningGroundFromIgnite = false
			end
		end
	end
```
Expand out assumption of always fire.

**Comment by QuickStick123** [comment]
This could use a breakdown but It doesn't belong to any skill so I don't know where it would go the calculations are fine though and interacts correctly with other burning ground.

**Review by Wires77 — APPROVED** [review]
I think this and decay should probably be refactored into a generic method, but this looks okay for now.

**Pull request merged**